File: API/Entities/Billiards/BilliardsMatch.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace API.Entities.Billiards
{
    [Table("BilliardsMatch")]
    public class BilliardsMatch
    {
        public int Id { get; set; }
        public AppUser AppUserWin { get; set; }
        public int WinUserId { get; set; }
        public AppUser AppUserLose { get; set; }
        public int LoseUserId { get; set; }
        public BilliardsMatchType BilliardsMatchType { get; set; }
        public int TypeId { get; set; }
        public BilliardsMode BilliardsMode { get; set; }
        public int ModeId { get; set; }
        public Season Season { get; set; }
        public int SeasonNumberId { get; set; }
        public Tournament Tournament { get; set; }
        public int TournamentId { get; set; }

        public int WinnerWins { get; set; }
        public int LoserWins { get; set; }
        public int TotalGamesPlayed { get; set; }


    }
}

File: client/src/app/_models/billiards/BilliardsMatch.ts
export interface BilliardsMatch
{
    id: number;
    winUserId: number;
    loseUserId: number;
    winPhotoUrl: string;
    losePhotoUrl: string;
    typeId: number;
    modeId: number;
    seasonNumberId: number;
    tournamentId: number;
    winnerWins: number;
    loserWins: number;
    totalGamesPlayed: number;
    datePlayed: Date;
}
